src/p2p/net_node.inl

```diff
@@ -2225,6 +2225,16 @@ namespace nodetool
       return 1;
     }
 
+    for (auto const& zone : m_network_zones)
+    {
+      if(arg.node_data.peer_id == zone.second.m_config.m_peer_id)
```

Comment:
Have you observed this happening or was this found while auditing/scanning the code? I can't think of a way to trigger this behavior due to how the connections are made.

Reply:
It was happening to me, repeatedly, while connected to hyc's randomx testnet.

Reply:
You connected to that network while enabling I2P and/or Tor? Because it looks like the issue _may_ have been that the existing check continued after the `if` block.

I avoided this type of check because I was concerned about metadata leakage. I could receive your Tor `peer_id`, then test your IPv4 connection to see what happens.

Reply:
I was connected by clearnet. Good point about testing, I'll change it to test only the remote end's zone.